pkg/services/ngalert/provisioning/persist.go

```diff
@@ -24,3 +24,20 @@ type ProvisioningStore interface {
 type TransactionManager interface {
 	InTransaction(ctx context.Context, work func(ctx context.Context) error) error
 }
+
+type ProvenanceOrgAdapter struct {
+	Inner models.ProvisionableInOrg
```

Comment:
Is the usual implementation of this a pointer or struct type?

Reply:
Struct. I'm working on a PR to drop this type entirely, since It's being used in way more places than I expected. Originally, I was expecting this to be a one-off shim that connects two interfaces.

Reply:
I will keep this how it is, since this object is very short-lived, but I'll end up dropping it.

Reply:
PR in question: https://github.com/grafana/grafana/pull/48137

that PR deletes this type entirely